python/amici/sbml_preprocessing.py

```diff
@@ -0,0 +1,106 @@
+import libsbml
+
+
+def flatten_sbml(sbml: libsbml.Model):
+    """Reformulate parts of the SBML, in order to allow
+        the import of some SBML features.
+
+    Arguments:
+        sbml: Instance of libsbml.Model`.
+
+    Returns:
+
+    Raises:
+    """
+
+    process_rate_rules(sbml)
+    process_species_assignment_rules(sbml)
+
+
+def process_rate_rules(sbml: libsbml.Model):
+    """ Reformulate rate rules by an reaction of the form
+     R: -> X with rate f(x).
+
+    Arguments:
+        sbml: Instance of libsbml.Model`.
+
+    Returns:
+
+    Raises:
+    """
+
+    # loop backwards over rules, so that one can delete rules inside the loop!
+    for i in range(sbml.num_rules - 1, -1, -1):
```

Comment:
does `reversed(sbml.getListOfRules())`/`reversed(list(sbml.getListOfRules()))`/`reversed(list(rule for rule in sbml.getListOfRules()))` work here? Seems more pythonic.

Reply:
good point!